src/file_pipe_log/pipe.rs

```diff
@@ -349,25 +242,39 @@ impl<F: FileSystem> SinglePipe<F> {
         reader.read(handle)
     }
 
+    /// Returns a shared [`LogFd`] for the specified file sequence number.
+    fn get_fd(&self, file_seq: FileSeq) -> Result<Arc<F::Handle>> {
```

Comment:
Use the old implementation and placement.

Reply:
"placement" too, so the diff is shorter and history is clean.

Reply:
Placement means put this function before `rotate_imp`, where it was before this PR.